src/Extractor/ContentExtractor.php

```diff
@@ -685,31 +685,39 @@ public function getNextPageUrl(): ?string
     /**
      * Validate and convert a date to the W3C format.
      *
-     * @param string $date
-     *
      * @return string|null Formatted date using the W3C format (Y-m-d\TH:i:sP) OR null if the date is badly formatted
      */
     public function validateDate(?string $date): ?string
     {
-        $date = (string) $date;
+        if (null === $date) {
+            return null;
+        }
+
+        // Presumably, this can return false on older PHP versions?
         $parseDate = (array) date_parse($date);
 
         // If no year has been found during date_parse, we nuke the whole value
         // because the date is invalid
-        if (null !== $date && false === $parseDate['year']) {
+        $parsedYear = $parseDate['year'] ?? false;
```

Comment:
I added it for when `date_parse` returns `false` and is cast to array. But it is not clear if that can actually happen. I did not manage to trigger it when I [tried passing it various invalid values](https://3v4l.org/6CQiI). So perhaps the [docs](https://www.php.net/manual/en/function.date-parse.php) are incorrect and it can only return `array` and the cast is pointless.

Reply:
I never got a chance to got a `false` either but it was reported by PHPStan.